src/PkgAuthentication.jl

```diff
@@ -175,8 +175,31 @@ end
 Base.show(io::IO, s::NoAuthentication) = print(io, "NoAuthentication($(s.server))")
 
 function step(state::NoAuthentication)::Union{RequestLogin, Failure}
-    challenge = Random.randstring(32)
     output = IOBuffer()
+    # Try device flow first
+    response = Downloads.request(
+        joinpath(state.server, "dex/device/code"),
+        method = "POST",
+        input = IOBuffer("client_id=device&scope=openid email profile offline_access"),
```

Comment:
The client_id should be configurable (maybe via env?), with "device" as the default. We may not be able to use the same client id everywhere, or it may need to be changed. If that is isolated as a separate method, it could be used at all the places client_id is required.

Reply:
Done in https://github.com/JuliaComputing/PkgAuthentication.jl/pull/42/commits/79e417d17ba77db3945a743853aa08d7a126385e